src/paginate.ts

```diff
@@ -175,6 +178,144 @@ export const DEFAULT_MAX_LIMIT = 100
 export const DEFAULT_LIMIT = 20
 export const NO_PAGINATION = 0
 
+type ColumnProperties = { propertyPath?: string; propertyName: string }
+
+function getPropertiesByColumnName(column: string): ColumnProperties {
+    const propertyPath = column.split('.')
+    return propertyPath.length > 1
+        ? {
+              propertyPath: propertyPath[0],
+              propertyName: propertyPath.slice(1).join('.'), // the join is in case of an embedded entity
+          }
+        : { propertyName: propertyPath[0] }
+}
+
+function extractVirtualProperty(
```

Comment:
same, alternatively let's move all filter helper functions already to it's own file

./paginate/filter.ts

then it's okay to keep the broader names